src/hypervisor.tex

```diff
@@ -404,6 +404,51 @@ \subsection{Hypervisor Counter-Enable Register ({\tt hcounteren})}
 when V=1.
 Hence, they are effectively \warl\ fields.
 
+\subsection{Hypervisor time delta registers ({\tt htimedelta}, {\tt htimedeltah})}
+
+The {\tt htimedelta} CSR is a VSXLEN-bit read/write register
+that contains a delta between the hardware value of the {\tt time} CSR
+and the value returned in VS/VU-mode.  That is, reading the {\tt time}
+or {\tt timeh} CSRs, or executing the {\tt RDTIME} instruction returns the
+value of the hardware counter plus the contents of {\tt htimedelta}.
```

Comment:
should this not be "returns the value of the hardware counter minus the contents of htimedelta" ??

Otherwise looks good to me.

Reply:
It's arbitrary, and we can define it either way, it's just a matter of whether software negates the value before writing it to htimedelta.

Reply:
I looked up how the delta registers were defined in the older versions of the specs, and we also defined it like Paolo did: the delta is added, not subtracted.

Reply:
I am fine with both approaches but we should have clear documentation of HTIMEDELTA CSR.

Maybe Replace
".... contains a delta between the hardware value of the {\tt time} CSR and the value returned in VS/VU-mode."
With
".... contains difference between the {\tt time} CSR value read in VS/VU-mode and the hardware value of the {\tt time} CSR."